e2e-tests/tfm/cypress/e2e/journeys/case-amendments/facility-end-date/feature-flag-enabled/gef-deal-add-bank-review-date.spec.js

```diff
@@ -0,0 +1,147 @@
+
+    beforeEach(() => {
+      cy.login(PIM_USER_1);
+      cy.visit(relative(`/case/${dealId}/facility/${facility._id}`));
+    });
+
+    after(() => {
+      cy.deleteDeals(dealId, ADMIN);
+    });
+
+    it('should display the default blank facility end date fields in the Details tab', () => {
+      facilityPage.facilityIsUsingFacilityEndDate().should('have.text', '-');
+      facilityPage.facilityFacilityEndDate().should('have.text', '-');
+      facilityPage.facilityBankReviewDate().should('not.exist');
+    });
+
+    it('should navigate to the bank review date page after selecting "No" to "Has the bank provided a facility end date"', () => {
+      amendmentsPage.navigateToIsUsingFacilityEndDatePage({ startNewAmendment: true });
+      amendmentsPage.isUsingFacilityEndDateYes().should('not.be.checked');
+      amendmentsPage.isUsingFacilityEndDateNo().should('not.be.checked');
+
+      amendmentsPage.continueAmendment().click();
+      amendmentsPage.errorSummary().contains('Select if the bank has provided an end date for this facility');
+
+      amendmentsPage.isUsingFacilityEndDateNo().click();
+      amendmentsPage.continueAmendment().click();
+
+      cy.url().should('contain', 'bank-review-date');
+      amendmentsPage.amendmentCurrentBankReviewDate().should('have.text', 'Not provided');
+      amendmentsPage.amendmentBankReviewDateDetails().should('exist');
+    });
+
+    it('should return expected errors when entering an incorrect bank review date', () => {
+      amendmentsPage.navigateToIsUsingFacilityEndDatePage();
+      cy.url().should('contain', 'bank-review-date');
+      amendmentsPage.amendmentBankReviewDateDayInput().clear().type(dateConstants.sixYearsOneDayDay);
+      amendmentsPage.amendmentBankReviewDateMonthInput().clear().type(dateConstants.sixYearsOneDayMonth);
+      amendmentsPage.amendmentBankReviewDateYearInput().clear().type(dateConstants.sixYearsOneDayYear);
+
+      amendmentsPage.continueAmendment().click();
+      amendmentsPage.errorSummary().contains('Bank review date cannot be greater than 6 years in the future');
+    });
+
+    it('should continue to the facility value page if the bank review date is valid and the facility value also needs changing', () => {
+      amendmentsPage.navigateToIsUsingFacilityEndDatePage({ changeFacilityValue: true });
+      amendmentsPage.continueAmendment().click();
+
+      cy.url().should('contain', 'bank-review-date');
+      amendmentsPage.amendmentBankReviewDateDayInput().clear().type(dateConstants.todayDay);
+      amendmentsPage.amendmentBankReviewDateMonthInput().clear().type(dateConstants.todayMonth);
+      amendmentsPage.amendmentBankReviewDateYearInput().clear().type(dateConstants.todayYear);
+      amendmentsPage.continueAmendment().click();
+
+      cy.url().should('contain', 'facility-value');
+    });
+
+    it('should continue to `Check your answers` page if the bank review date is valid and only the cover end date is being changed', () => {
```

Comment:
this can be made shorter with a describe and then an it('should continue to check your answers page').  Should have a tech debt ticket for these.  Will be easier to follow

Reply:
I'm going to break down this test file -- good idea. I've broken it down so we actually include in the describe block what the describe block is desribing to keep things DRY